File: Case_Latency_Auditor.py
def analizar_picos_latencia(tiempos, umbral_critico):
    if not tiempos:
        return False 

    promedio = sum(tiempos) / len(tiempos)
    print(f"--- Análisis de Red ---")
    print(f"Promedio actual: {promedio:.2f} ms")
    print(f"Umbral crítico: {umbral_critico} ms")

    promedio_excedido = promedio > umbral_critico

    pico_detectado = False
    for t in tiempos:
        if t >= (promedio * 3):
            pico_detectado = True
            print(f"!Pico sospechoso detectado!: {t} ms")
            break 

    disparar_alerta = promedio_excedido or pico_detectado
    
    return disparar_alerta

datos_latencia = [120, 135, 110, 450, 125, 140] 
umbral = 200

alerta_activa = analizar_picos_latencia(datos_latencia, umbral)

if alerta_activa:
    print("ALERTA: Se han detectado irregularidades en la latencia.")
else:
    print("Sistema estable: Latencia dentro de los parámetros.")

File: CaseLatencyAuditor.js
function analizarPicosLatencia(tiempos, umbralCritico) {
    if (!tiempos || tiempos.length === 0) {
        return false;
    }

    const suma = tiempos.reduce((acumulador, valorActual) => acumulador + valorActual, 0);
    const promedio = suma / tiempos.length;

    console.log("--- Análisis de Red ---");
    console.log(`Promedio actual: ${promedio.toFixed(2)} ms`);
    console.log(`Umbral crítico: ${umbralCritico} ms`);

    const promedioExcedido = promedio > umbralCritico;

    let picoDetectado = false;
    for (let t of tiempos) {
        if (t >= (promedio * 3)) {
            picoDetectado = true;
            console.log(`!Pico sospechoso detectado!: ${t} ms`);
            break;
        }
    }

    const dispararAlerta = promedioExcedido || picoDetectado;
    
    return dispararAlerta;
}

const datosLatencia = [120, 135, 110, 450, 125, 140];
const umbral = 200;

const alertaActiva = analizarPicosLatencia(datosLatencia, umbral);

if (alertaActiva) {
    console.log("ALERTA: Se han detectado irregularidades en la latencia.");
} else {
    console.log("Sistema estable: Latencia dentro de los parámetros.");
}